lib/auth/auth.go

```diff
@@ -591,6 +593,53 @@ func (s *AuthServer) generateUserCert(req certRequest) (*certs, error) {
 	return &certs{ssh: sshCert, tls: tlsCert}, nil
 }
 
+// CheckOrSetKubeCluster returns a valid kubernetes cluster name.
+// If kc is set, it's validated against known clusters.
+// If kc is empty, one of the known cluster names is returned instead.
+func CheckOrSetKubeCluster(kc string, pg services.ProxyGetter) (string, error) {
+	proxies, err := pg.GetProxies()
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	// If k8s cluster name was provided, validate that it's a registered
+	// cluster.
+	if kc != "" {
+		for _, p := range proxies {
+			pp, ok := p.(*services.ServerV2)
+			if !ok {
+				continue
+			}
+			for _, pkc := range pp.Spec.KubernetesClusters {
+				if pkc == kc {
+					return kc, nil
+				}
+			}
+		}
+		return "", trace.BadParameter("kubernetes cluster %q is not registered in this teleport cluster; you can list registered kubernetes clusters using 'tsh kube clusters'", kc)
```

Comment:
Has the handler for `tsh kube` been implemented yet? I'd be cautious about adding this message in ahead of time.

Reply:
Good point, `tsh kube ...` is not implemented just yet.
I made the `--kube-cluster` flag hidden for now, so users shouldn't hit this code branch unless they read the code.